syntax = "proto3";

package bookowl;

option go_package = "github.com/jphacks/B_2109/server/api";

import "google/protobuf/timestamp.proto";

service ReadEvent{
  // Registering BookEvent with bookmark ID
  rpc Register(ReadEventRegisterRequest) returns (ReadEventRegisterResponse);
}

message ReadEventRegisterRequest {
  ReadEventInfo read_event_info = 1;
  string bookmark_id = 2;
}

message ReadEventInfo {
  google.protobuf.Timestamp read_start_time = 1;
  google.protobuf.Timestamp read_end_time = 2;
  int64 read_start_position = 3;
  int64 read_end_position = 4;
}

message ReadEventRegisterResponse {
  string read_event_id = 1;
  google.protobuf.Timestamp time = 2;
}
